[7.0] add asset management modules

luc-demeyer opened the pull request.

luc-demeyer committed:
add asset management modules

gurneyalex commented:
That's a huge PR. I've made some changes in #53 to :+1: 
- change the travis config to add the new dependencies
- fix the flake8 warnings

@luc-demeyer you may want to pull these changes in your branch if you want to continue working on that PR.

The tests are still red on #53 because of problems in the demo data of account_asset_management (https://travis-ci.org/OCA/account-financial-tools/jobs/36747656) 

This could be either:
- a problem in your code
- a problem with another module being tested together with this one.

I have not looked further. If it is the former, please make the changes, otherwise, please complain here and I shall adapt the Travis configuration.

luc-demeyer commented:
Alexandre,

FIXING the flake8 warnings as you did breaks the module.

You should NOT remove the ‘unused variables’ in none of the report_xls dependent modules because flake8 is not capable of understanding the ‘python name space’ trick done by report_xls.

Concerning the demo data : 

These are still the ‘demo data’ from the OpenERP account_asset module that I used as a starting point. 

In the meanwhile this module has been completely refactored (I’ll add the changelog into the doc folder) and all of these tests should be completely reviewed.

The good news is that we are running this module at three customer sites in production hence that compensates for the lack of a valid test bed.

I’ll make a new PR with the demo data commented out. I hope that you can find a way to fix the (in this particular case) incorrect flake8 warnings.

Regards,

Luc

 http://www.noviat.com/ www.noviat.com

Rusatiralaan 1, 1083 Brussel

+32 2 808 86 38

From: Alexandre Fayolle [mailto:notifications@github.com] 
Sent: woensdag 1 oktober 2014 10:32
To: OCA/account-financial-tools
Cc: luc-demeyer
Subject: Re: [account-financial-tools] add asset management modules (#51)

That's a huge PR. I've made some changes in  https://github.com/OCA/account-financial-tools/pull/53 #53 to   https://assets-cdn.github.com/images/icons/emoji/unicode/1f44d.png 
-   change the travis config to add the new dependencies
-   fix the flake8 warnings

@luc-demeyer https://github.com/luc-demeyer  you may want to pull these changes in your branch if you want to continue working on that PR.

The tests are still red on #53 https://github.com/OCA/account-financial-tools/pull/53  because of problems in the demo data of account_asset_management (https://travis-ci.org/OCA/account-financial-tools/jobs/36747656) 

This could be either:
-   a problem in your code
-   a problem with another module being tested together with this one.

I have not looked further. If it is the former, please make the changes, otherwise, please complain here and I shall adapt the Travis configuration. 

—
Reply to this email directly or view it on GitHub https://github.com/OCA/account-financial-tools/pull/51#issuecomment-57433911 .  https://github.com/notifications/beacon/7706945__eyJzY29wZSI6Ik5ld3NpZXM6QmVhY29uIiwiZXhwaXJlcyI6MTcyNzc3MTUyOCwiZGF0YSI6eyJpZCI6NDQzNDYwNjR9fQ==--50685c7366731b3238e0bc1f21bbcba9eb0beeeb.gif

luc-demeyer committed:
update V7 account_asset_management module

gurneyalex commented:
OK, my bad, I'll propose another way then. Feel free to cherry pick the
commit updating the travis configuration in the meantime.

## 

Alexandre

On 01/10/2014 11:30, luc-demeyer wrote:

> Alexandre,
> 
> FIXING the flake8 warnings as you did breaks the module.
> 
> You should NOT remove the ‘unused variables’ in none of the report_xls
> dependent modules because flake8 is not capable of understanding the
> ‘python name space’ trick done by report_xls.
> 
> Concerning the demo data :
> 
> These are still the ‘demo data’ from the OpenERP account_asset module
> that I used as a starting point.
> 
> In the meanwhile this module has been completely refactored (I’ll add
> the changelog into the doc folder) and all of these tests should be
> completely reviewed.
> 
> The good news is that we are running this module at three customer
> sites in production hence that compensates for the lack of a valid
> test bed.
> 
> I’ll make a new PR with the demo data commented out. I hope that you
> can find a way to fix the (in this particular case) incorrect flake8
> warnings.
> 
> Regards,
> 
> Luc
> 
> http://www.noviat.com/ www.noviat.com
> 
> Rusatiralaan 1, 1083 Brussel
> 
> +32 2 808 86 38
> 
> From: Alexandre Fayolle [mailto:notifications@github.com]
> Sent: woensdag 1 oktober 2014 10:32
> To: OCA/account-financial-tools
> Cc: luc-demeyer
> Subject: Re: [account-financial-tools] add asset management modules (#51)
> 
> That's a huge PR. I've made some changes in
> https://github.com/OCA/account-financial-tools/pull/53 #53 to
> https://assets-cdn.github.com/images/icons/emoji/unicode/1f44d.png
> - change the travis config to add the new dependencies
> - fix the flake8 warnings
> 
> @luc-demeyer https://github.com/luc-demeyer you may want to pull
> these changes in your branch if you want to continue working on that PR.
> 
> The tests are still red on #53
> https://github.com/OCA/account-financial-tools/pull/53 because of
> problems in the demo data of account_asset_management
> (https://travis-ci.org/OCA/account-financial-tools/jobs/36747656)
> 
> This could be either:
> - a problem in your code
> - a problem with another module being tested together with this one.
> 
> I have not looked further. If it is the former, please make the
> changes, otherwise, please complain here and I shall adapt the Travis
> configuration.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/OCA/account-financial-tools/pull/51#issuecomment-57433911
> .
> https://github.com/notifications/beacon/7706945__eyJzY29wZSI6Ik5ld3NpZXM6QmVhY29uIiwiZXhwaXJlcyI6MTcyNzc3MTUyOCwiZGF0YSI6eyJpZCI6NDQzNDYwNjR9fQ==--50685c7366731b3238e0bc1f21bbcba9eb0beeeb.gif
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/OCA/account-financial-tools/pull/51#issuecomment-57439644.

## 

Alexandre Fayolle
Chef de Projet
Tel : + 33 (0)4 79 26 57 94

Camptocamp France SAS
Savoie Technolac, BP 352
73377 Le Bourget du Lac Cedex
http://www.camptocamp.com

gurneyalex commented:
@luc-demeyer I've updated the branch in PR #53:
1. rebased on top of your latest commit on the present PR
2. changed my handling of "unused variable" warnings by using an inline directive to selectively disable the warning on the affected lines
3. remove really unused symbols

The Travis tests for that branch are still red: https://travis-ci.org/OCA/account-financial-tools/jobs/36946058 (failure to load account_asset_management/account_asset_demo.xml)

luc-demeyer commented:
Thanks, 

I’ll have a look at the changes you have made and I’ll fix the errors reported by the travis tests.

Regards,

Luc

 http://www.noviat.com/ www.noviat.com

Rusatiralaan 1, 1083 Brussel

+32 2 808 86 38

From: Alexandre Fayolle [mailto:notifications@github.com] 
Sent: vrijdag 3 oktober 2014 10:37
To: OCA/account-financial-tools
Cc: luc-demeyer
Subject: Re: [account-financial-tools] add asset management modules (#51)

@luc-demeyer https://github.com/luc-demeyer  I've updated the branch in PR #53 https://github.com/OCA/account-financial-tools/pull/53 :
1.  rebased on top of your latest commit on the present PR
2.  changed my handling of "unused variable" warnings by using an inline directive to selectively disable the warning on the affected lines
3.  remove really unused symbols

The Travis tests for that branch are still red: https://travis-ci.org/OCA/account-financial-tools/jobs/36946058 (failure to load account_asset_management/account_asset_demo.xml) 

—
Reply to this email directly or view it on GitHub https://github.com/OCA/account-financial-tools/pull/51#issuecomment-57769311 .  https://github.com/notifications/beacon/7706945__eyJzY29wZSI6Ik5ld3NpZXM6QmVhY29uIiwiZXhwaXJlcyI6MTcyNzk0NDYyMSwiZGF0YSI6eyJpZCI6NDQzNDYwNjR9fQ==--df0b44c937d87f560bdd91aee89073fc0f24e765.gif

luc-demeyer committed:
V7 add asset management with demo & yml

luc-demeyer committed:
add reporting_engine to 7.0 travis.yml

luc-demeyer committed:
7.0 - fix account_asset.yml

luc-demeyer commented:
Alexandre,

Thanks a lot for your help. 

I reviewed the changes you made and could remove two of the “# noqa unused variables” since not used in the XLS template (and hence should never have been there).

I also 
- ```
       fixed the error reported by Travis with the demo data
  ```
- ```
       added the reporting_engine repo to .travis.yml
  ```
- ```
       included some yml tests and a  couple of other small enhancements
  ```

I still have a travis error, cf. https://travis-ci.org/OCA/account-financial-tools/jobs/37104681 :

dropdb: database removal failed: ERROR:  database "openerp_test" does not exist (looks like an error in the travis maintainer tools).

Can you fix this ? 

In the meanwhile, the prerequisite level of the ‘report_xls’ module has been merged into the  reporting-engine 7.0 branch, hence Frederic can start the functional testing once this last travis error is fixed.

Regards,

Luc 

 http://www.noviat.com/ www.noviat.com

Rusatiralaan 1, 1083 Brussel

+32 2 808 86 38

From: Alexandre Fayolle [mailto:notifications@github.com] 
Sent: vrijdag 3 oktober 2014 10:37
To: OCA/account-financial-tools
Cc: luc-demeyer
Subject: Re: [account-financial-tools] add asset management modules (#51)

@luc-demeyer https://github.com/luc-demeyer  I've updated the branch in PR #53 https://github.com/OCA/account-financial-tools/pull/53 :
1.  rebased on top of your latest commit on the present PR
2.  changed my handling of "unused variable" warnings by using an inline directive to selectively disable the warning on the affected lines
3.  remove really unused symbols

The Travis tests for that branch are still red: https://travis-ci.org/OCA/account-financial-tools/jobs/36946058 (failure to load account_asset_management/account_asset_demo.xml) 

—
Reply to this email directly or view it on GitHub https://github.com/OCA/account-financial-tools/pull/51#issuecomment-57769311 .  https://github.com/notifications/beacon/7706945__eyJzY29wZSI6Ik5ld3NpZXM6QmVhY29uIiwiZXhwaXJlcyI6MTcyNzk0NDYyMSwiZGF0YSI6eyJpZCI6NDQzNDYwNjR9fQ==--df0b44c937d87f560bdd91aee89073fc0f24e765.gif

luc-demeyer committed:
update account_asset_management

florian-dacosta commented on account_asset_management/account_asset.py:
```diff
@@ -0,0 +1,1706 @@
+                        'parent_id', 'depreciation_line_ids'
+                    ], 20),
+                'account.asset.depreciation.line': (
+                    _get_assets_from_dl, ['amount', 'init', 'move_id'], 20),
```
Hello, 
I think it should be 'init_entry' instead of 'init' since there is no field named 'init' in account.asset.depreciation.line.

Thanks for your great work!

luc-demeyer replied:
Florian,

I am impressed by the thoroughness of your code review. 

It should indeed be ‘init_entry’ in stead of ‘init’ and you can find this out only by reading the code (since there are no cases in the current version of this module where the ‘init_entry’ field can be  modified without any of the other trigger fields). 

I have changed ‘init’ into ‘init_entry’ in the meanwhile.

Thanks,

Luc

 http://www.noviat.com/ www.noviat.com

Rusatiralaan 1, 1083 Brussel

+32 2 808 86 38

From: florian-dacosta [mailto:notifications@github.com] 
Sent: maandag 27 oktober 2014 18:32
To: OCA/account-financial-tools
Cc: luc-demeyer
Subject: Re: [account-financial-tools] add asset management modules (#51)

In account_asset_management/account_asset.py:

> -                    _get_assets,
> -                    ['purchase_value', 'salvage_value', 'parent_id'], 10),
> -            },
> -            help="This amount represent the initial value of the asset."),
> -        'value_residual': fields.function(
> -            _residual, method=True,
> -            digits_compute=dp.get_precision('Account'),
> -            string='Residual Value',
> -            store={
> -                'account.asset.asset': (
> -                    _get_assets, [
> -                        'purchase_value', 'salvage_value',
> -                        'parent_id', 'depreciation_line_ids'
> -                    ], 20),
> -                'account.asset.depreciation.line': (
> -                    _get_assets_from_dl, ['amount', 'init', 'move_id'], 20),

Hello, 
I think it should be 'init_entry' instead of 'init' since there is no field named 'init' in account.asset.depreciation.line.

Thanks for your great work!

—
Reply to this email directly or view it on GitHub https://github.com/OCA/account-financial-tools/pull/51/files#r19421666 .  https://github.com/notifications/beacon/AHWZQdx9c_MsPVX5O2OKXalXCRNFl6uOks5nHnkogaJpZM4CpKrQ.gif

florian-dacosta replied:
Thank you for the quick fix!
To be honest, it is not just reading, I am trying the module and I need to use this field, that's why I saw the problem ;)

luc-demeyer replied:
Can I assume that you are developping a script to import an existing asset table with as a consequence entries from closed fiscal years ? 

If yes, that would be a great add-on module or alternatively a wizard that we merge into the existing module.

Regards,

Luc

 http://www.noviat.com/ www.noviat.com

Rusatiralaan 1, 1083 Brussel

+32 2 808 86 38

From: florian-dacosta [mailto:notifications@github.com] 
Sent: dinsdag 28 oktober 2014 10:00
To: OCA/account-financial-tools
Cc: luc-demeyer
Subject: Re: [account-financial-tools] add asset management modules (#51)

In account_asset_management/account_asset.py:

> -                    _get_assets,
> -                    ['purchase_value', 'salvage_value', 'parent_id'], 10),
> -            },
> -            help="This amount represent the initial value of the asset."),
> -        'value_residual': fields.function(
> -            _residual, method=True,
> -            digits_compute=dp.get_precision('Account'),
> -            string='Residual Value',
> -            store={
> -                'account.asset.asset': (
> -                    _get_assets, [
> -                        'purchase_value', 'salvage_value',
> -                        'parent_id', 'depreciation_line_ids'
> -                    ], 20),
> -                'account.asset.depreciation.line': (
> -                    _get_assets_from_dl, ['amount', 'init', 'move_id'], 20),

Thank you for the quick fix!
To be honest, it is not just reading, I am trying the module and I need to use this field, that's why I saw the problem ;)

—
Reply to this email directly or view it on GitHub https://github.com/OCA/account-financial-tools/pull/51/files#r19458531 .  https://github.com/notifications/beacon/AHWZQYhIeyS8awSUIvGsqQvyssBZQB6tks5nH1KGgaJpZM4CpKrQ.gif

florian-dacosta replied:
Actually I just import a custom excel file with old and current assets into Odoo.
I am not sure to understand what you mean, since there is already a script to migrate from account_asset to account_asset_management.
I would be glad to contribute what I did but it is ultra specific and made in ooor...

luc-demeyer replied:
Florian, 

I indeed made a script to migrate a couple of our customers from the old to the new version of the asset mgt module.

What I mean is that any customer who starts managing his assets in openerp/odoo will either have to type them in manually or alternatively export them from e.g. an excel or csv file.

Hence if you make an ORM script to import old and current assets from an XLS file, than we could document the XLS (or probably csv) format  and adding an import wizard to load assets from this documented csv format.

We do have a customer project where we have to load 700+ assets into odoo hence it would be interesting if I could reuse the script that you are working on.

Regards,

Luc

 http://www.noviat.com/ www.noviat.com

Rusatiralaan 1, 1083 Brussel

+32 2 808 86 38

From: florian-dacosta [mailto:notifications@github.com] 
Sent: dinsdag 28 oktober 2014 11:15
To: OCA/account-financial-tools
Cc: luc-demeyer
Subject: Re: [account-financial-tools] add asset management modules (#51)

In account_asset_management/account_asset.py:

> -                    _get_assets,
> -                    ['purchase_value', 'salvage_value', 'parent_id'], 10),
> -            },
> -            help="This amount represent the initial value of the asset."),
> -        'value_residual': fields.function(
> -            _residual, method=True,
> -            digits_compute=dp.get_precision('Account'),
> -            string='Residual Value',
> -            store={
> -                'account.asset.asset': (
> -                    _get_assets, [
> -                        'purchase_value', 'salvage_value',
> -                        'parent_id', 'depreciation_line_ids'
> -                    ], 20),
> -                'account.asset.depreciation.line': (
> -                    _get_assets_from_dl, ['amount', 'init', 'move_id'], 20),

Actually I just import a custom excel file with old and current assets into Odoo.
I am not sure to understand what you mean, since there is already a script to migrate from account_asset to account_asset_management.
I would be glad to contribute what I did but it is ultra specific and made in ooor...

—
Reply to this email directly or view it on GitHub https://github.com/OCA/account-financial-tools/pull/51/files#r19461440 .  https://github.com/notifications/beacon/AHWZQfum8cR_1I9UxJD7GwfjttLdEop6ks5nH2QrgaJpZM4CpKrQ.gif

florian-dacosta commented on account_asset_management/account_asset.py:
```diff
@@ -0,0 +1,1706 @@
+                        'parent_id', 'depreciation_line_ids'
+                    ], 30),
+                'account.asset.depreciation.line': (
+                    _get_assets_from_dl, ['amount', 'init', 'move_id'], 30),
```
same comment here, It should be 'init_entry' instead of 'init'

luc-demeyer replied:
Indeed.

I have fixed this in the meanwhile.

Luc

 http://www.noviat.com/ www.noviat.com

Rusatiralaan 1, 1083 Brussel

+32 2 808 86 38

From: florian-dacosta [mailto:notifications@github.com] 
Sent: maandag 27 oktober 2014 18:35
To: OCA/account-financial-tools
Cc: luc-demeyer
Subject: Re: [account-financial-tools] add asset management modules (#51)

In account_asset_management/account_asset.py:

> -                    ], 20),
> -                'account.asset.depreciation.line': (
> -                    _get_assets_from_dl, ['amount', 'init', 'move_id'], 20),
> -            }),
> -        'value_depreciated': fields.function(
> -            _depreciated, method=True,
> -            digits_compute=dp.get_precision('Account'),
> -            string='Depreciated Value',
> -            store={
> -                'account.asset.asset': (
> -                    _get_assets, [
> -                        'purchase_value', 'salvage_value',
> -                        'parent_id', 'depreciation_line_ids'
> -                    ], 30),
> -                'account.asset.depreciation.line': (
> -                    _get_assets_from_dl, ['amount', 'init', 'move_id'], 30),

same comment here, It should be 'init_entry' instead of 'init'

—
Reply to this email directly or view it on GitHub https://github.com/OCA/account-financial-tools/pull/51/files#r19421886 .  https://github.com/notifications/beacon/AHWZQT_Rz1iNHkoVarrHYDxraLgyuPSjks5nHnnBgaJpZM4CpKrQ.gif

luc-demeyer committed:
fix account_asset_management

luc-demeyer committed:
refactor code to allow move and move_line vals change via inherit

ecino commented:
:+1:  

We use the module in production and it works well.

ecino commented on account_asset_management/account_asset_invoice.py:
```diff
@@ -0,0 +1,114 @@
+class account_invoice(orm.Model):
+    _inherit = 'account.invoice'
+
+    def action_number(self, cr, uid, ids, *args):
```
This is not the correct way of defining the function. You should replace it by :
    def action_number(self, cr, uid, ids, context=None):
Or
    def action_number(self, cr, uid, ids, **kwargs):

luc-demeyer replied:
Indeed.

This is a bug coming from the standard odoo ‘account_asset’ module (I checked and still not fixed in V8) and probably never remarked/reported because this method is usually called via the workflow. 

I have corrected this and submitted the fix.

Regards,

Luc

 http://www.noviat.com/ www.noviat.com

Rusatiralaan 1, 1083 Brussel

+32 2 808 86 38

From: ecino [mailto:notifications@github.com] 
Sent: vrijdag 31 oktober 2014 16:26
To: OCA/account-financial-tools
Cc: luc-demeyer
Subject: Re: [account-financial-tools] add asset management modules (#51)

In account_asset_management/account_asset_invoice.py:

> +#    GNU Affero General Public License for more details.
> +#
> +#    You should have received a copy of the GNU Affero General Public License
> +#    along with this program. If not, see http://www.gnu.org/licenses/.
> +#
> +##############################################################################
> +
> +from openerp.osv import fields, orm
> +import logging
> +_logger = logging.getLogger(__name__)
> +
> +
> +class account_invoice(orm.Model):
> -    _inherit = 'account.invoice'
>   +
> -    def action_number(self, cr, uid, ids, *args):

This is not the correct way of defining the function. You should replace it by :
def action_number(self, cr, uid, ids, context=None):
Or
def action_number(self, cr, uid, ids, **kwargs):

—
Reply to this email directly or view it on GitHub https://github.com/OCA/account-financial-tools/pull/51/files#r19672864 .  https://github.com/notifications/beacon/AHWZQXIQoLVJTln9BnAQB0A4xcj4QB-Xks5nI6GBgaJpZM4CpKrQ.gif

luc-demeyer committed:
asset mgt, fix action_number

sbidoul commented:
@luc-demeyer can you cherry-pick this commit https://github.com/acsone/account-financial-tools/commit/155cb2a6417b938cb9d93c7ae55bc538476325be ? It will silence a pylint error.

sbidoul commented:
@luc-demeyer otherwise the module works well in my tests. Thanks for this work!

So :+1: for me too (test). I can't say I read every single line of code, but overall it looks quite clean.

When this is merged, I can work on the 8.0 forward port.

luc-demeyer commented:
Stephane,  

No need to work on an 8.0 port.

That work is already done and I’ll push the 8.0 PR immediately after the merge of the 7.0 (the same applies for the 6.1 backport).

If have attached the 80 port to this mail so you can test.

If you have time/resources than it would be great if you could work on the v8 ‘BI’ part of the module since I haven’t focused on that part yet (no hard customer requirements but nice for demos).

Regards,

Luc

 http://www.noviat.com/ www.noviat.com

Rusatiralaan 1, 1083 Brussel

+32 2 808 86 38

From: Stéphane Bidoul (ACSONE) [mailto:notifications@github.com] 
Sent: woensdag 19 november 2014 18:34
To: OCA/account-financial-tools
Cc: luc-demeyer
Subject: Re: [account-financial-tools] add asset management modules (#51)

@luc-demeyer https://github.com/luc-demeyer  otherwise the module works well in my tests. Thanks for this work!

So   https://assets-cdn.github.com/images/icons/emoji/unicode/1f44d.png for me too (test). I can't say I read every single line of code, but overall it looks quite clean.

When this is merged, I can work on the 8.0 forward port.

—
Reply to this email directly or view it on GitHub https://github.com/OCA/account-financial-tools/pull/51#issuecomment-63679674 .  https://github.com/notifications/beacon/AHWZQcsq2f-N5Do9iX_X6OvO7gQVPKTVks5nPMwTgaJpZM4CpKrQ.gif

luc-demeyer committed:
pylint

luc-demeyer commented:
Thanks Stéphane,

I have merged your improvement in the PR.

Regards,

Luc

 http://www.noviat.com/ www.noviat.com

Rusatiralaan 1, 1083 Brussel

+32 2 808 86 38

From: Stéphane Bidoul (ACSONE) [mailto:notifications@github.com] 
Sent: woensdag 19 november 2014 17:20
To: OCA/account-financial-tools
Cc: luc-demeyer
Subject: Re: [account-financial-tools] add asset management modules (#51)

@luc-demeyer https://github.com/luc-demeyer  can you cherry-pick this commit acsone@ https://github.com/acsone/account-financial-tools/commit/155cb2a6417b938cb9d93c7ae55bc538476325be 155cb2a ? It will silence a pylint error.

—
Reply to this email directly or view it on GitHub https://github.com/OCA/account-financial-tools/pull/51#issuecomment-63666301 .  https://github.com/notifications/beacon/AHWZQeWr95VIHZYtnd_fjF2kHwJhx3Cmks5nPLqzgaJpZM4CpKrQ.gif

sbidoul commented:
@luc-demeyer small detail: it seems github does not handle e-mail replies very well: the comments history becomes very hard to ready. It works better when replies are entered directly on the github website.

sbidoul commented:
@luc-demeyer I just noticed `'asset_report_xls' object has no attribute 'context'`error when running the Excel report on 7.0. Is it something in my setup or do you have it too?

luc-demeyer commented:
Stéphane, 

You need to update your local reporting engine OCA branch.
The latest version of report_xls is a prerequisite but since this one sits in another OCA branch you may run into such  problems. 

P.S. What i am missing is an improved ‘depends’ parameter of the **openerp**.py file so that you can specify a version, e.g. depends = [‘report_xls>=7.0.0.6’)]

sbidoul commented:
@luc-demeyer I had an old version of report_xls indeed. Thanks!

@gurneyalex can you or someone else remove the needs fixing flag from this PR?

sbidoul commented:
Hi @luc-demeyer. I'm afraid it's better to revert commit https://github.com/luc-demeyer/account-financial-tools/commit/4e6e2ea87d73f265ec7b8adafb230331116c198b because of https://github.com/OCA/maintainer-quality-tools/issues/141 which induced several people including me to make this potentially harmful change.

Also, while we are waiting for other reviews, could you re-do the 8.0 on top of the commits of the 7.0 branch? That will make the merge of 7.0 easier and will allow me to continue testing on 8.0. 

Thanks again for this contribution!

pedrobaeza commented:
I don't think this is needed, because this is not a keyword argument, so it is not needed to keep the argument name the same.

sbidoul commented:
@pedrobaeza still it's better to keep the original field name because any trailing argument can be used as a keyword argument:

``` python
def f(a, b=0):
    print a, b

f(a=1, b=2)
```

pedrobaeza commented:
Yeah, it can be. I have already changed that field name in a lot of places in OCA, so we have to pray that this doesn't generate any side effect...

luc-demeyer committed:
improvements asset mgt

luc-demeyer commented:
@sbidoul  I have reverted the 'pylint' commit and made a new commit with a couple of improvements:
- the initial balance flag is now automatically set for depreciation table entries in closed fiscal years
- asset date_start change is now reflected in the 'type create' entry of the depreciation table (via onchange)
- assets without depreciation table are now supported (required for asset that keep their value, e.g. land)
- table recalc bugfix

sbidoul commented:
Thanks @luc-demeyer could you re-do your 8.0 branch?

sbidoul commented:
@luc-demeyer can you add [7.0] in the pull request title?

fclementic2c commented:
Fonctionnal test Ok :+1:

sbidoul commented:
I confirm my :+1: https://github.com/OCA/account-financial-tools/pull/51#issuecomment-63679674

We did functional tests.

From a basic code review standpoint it would be nice to split account_asset.py in more manageable chunks, and remove the #vim lines but this is not blocking for me and can be left for future work.

fclementic2c commented:
Ok what's left to press on the button 'merged' ?

_camptocamp_

INNOVATIVE SOLUTIONS
BY OPEN SOURCE EXPERTS

_Frédéric Clementi_
Project Manager
Business Solutions

+41 21 619 10 41

www.camptocamp.com

2014-12-04 11:50 GMT+01:00 Stéphane Bidoul (ACSONE) <
notifications@github.com>:

> I confirm my [image: :+1:] #51 (comment)
> https://github.com/OCA/account-financial-tools/pull/51#issuecomment-63679674
> 
> We did functional tests.
> 
> From a basic code review standpoint it would be nice to split
> account_asset.py in more manageable chunks, and remove the #vim lines but
> this is not blocking for me and can be left for future work.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/OCA/account-financial-tools/pull/51#issuecomment-65614870
> .

florian-dacosta commented:
:+1:

sbidoul commented on account_asset_management/account_asset.py:
```diff
@@ -0,0 +1,1743 @@
+            "INNER JOIN account_asset_asset a ON dl.asset_id = a.id "
+            "WHERE a.id = %s AND dl.type='depreciate' "
+            "AND (dl.init_entry=TRUE OR dl.move_check=TRUE)",
+            (asset.id,))
```
@luc-demeyer The join on account_asset_asset is unncessary.

sbidoul committed:
[IMP] account_asset_management: performance improvement when recomputing the depreciation board

This commit removes an unncessary JOIN, and reduces the number
of computation of the depreciation values. It is not necessary to
recompute the depreciation values when computing the depreciation
board, only when posting depreciation lines.

luc-demeyer committed:
Merge pull request #3 from acsone/7.0-asset-management-perf-ldm-sbi

[IMP] account_asset_management: performance improvement when recomputing...
@sbidoul  : thanks for this improvement.

sbidoul commented:
@luc-demeyer, can you please review my last performance improvements PR to your branch. Without that the module is barely usable with a large number of assets.

One last thing: as I understand it, the override of account.fiscal.year forces a recomputation of all depreciation boards each time a new fiscal year is created. I'm not sure it is a good default use case, as when you have a large number of assets it is a very time consuming operation. What do you think?

sbidoul commented:
@luc-demeyer can you also do a git rebase?

luc-demeyer commented:
@sbidoul 
- I merged your PR 4 days ago.
- recompute at FY creation : 
  1) the code could be  optimised by limiting the recompute trigger to newly created FY that are extended/shortened FY's (I'll add a "TO DO" in the code so it get's on the list of things to improve).
  2) the _compute_entries method can be placed in a cron job running e.g. daily at night (cf. the "TO DO") in the code.

sbidoul commented:
@luc-demeyer I mean this PR: https://github.com/luc-demeyer/account-financial-tools/pull/4

luc-demeyer committed:
performance improvement + early removal

luc-demeyer committed:
early removal bugfix

sbidoul committed:
[IMP] account_asset_management: more tests (test_4 failing)

test_4 is failing as it is a scenario of having an init_entry
depreciation line in an active fiscal year. Next commits intend to
fix this.

sbidoul committed:
[IMP] account_asset_management: support moves with more than one asset depreciaion in the same move

Such moves are not generated by the module itself but may
exist when importing existing moves from another system.
This patch should otherwise not change the behaviour of the module.

florian-dacosta committed:
normal removal fix

luc-demeyer committed:
Merge pull request #5 from florian-dacosta/7.0-removal-fix

normal removal fix

luc-demeyer committed:
asset mgt - early removal asset_id filter

sbidoul committed:
[FIX] account_asset_management: support init entries in current fy

Fixes test_4. Handle situations where we have depreciation init entries
in the same fiscal year as real depreciation lines.

sbidoul commented:
@luc-demeyer I propose a couple of improvements related to migrating from other systems in https://github.com/luc-demeyer/account-financial-tools/pull/6

sbidoul committed:
[IMP] account_asset_management: simplify adjustment queries

Use depreciation lines instead of account moves since
there are now checks that ensure both are synchronised.

luc-demeyer committed:
Merge branch 'acsone-7.0-asset-management-fix3-sbi' into 7.0

luc-demeyer committed:
extra depr. line update controls

sbidoul committed:
[IMP] account_asset_management: replace official config option.

This prevents users clicking on the "enable assets management" option
and inadvertently installing account_asset.

luc-demeyer committed:
Merge pull request #7 from acsone/7.0-account_asset_management-config-sbi

[IMP] account_asset_management: replace official config option.

rdeheele commented:
Hi, 
what a huge work here!
Is it possible to merge it?
@luc-demeyer what about 8.0 PR that you have evoked in november?

thanks

luc-demeyer commented:
Romain,

We have already rolled it out to five sites and it’s working hence it can be merged.

I am just waiting on a committer of the account-financial-tools branch to push on the merge button (this can be done for both the 6.1 and 7.0 PR’s)

As soon as it’s merged I’ll update the 8.0 PR (since the last changes made on 7.0 are not yet forwarded ported to the 8.0 PR).

Regards,

Luc

 http://www.noviat.com/ www.noviat.com

Rusatiralaan 1, 1083 Brussel

+32 2 808 86 38

From: Romain Deheele [mailto:notifications@github.com] 
Sent: donderdag 8 januari 2015 16:02
To: OCA/account-financial-tools
Cc: luc-demeyer
Subject: Re: [account-financial-tools] [7.0] add asset management modules (#51)

Hi, 
what a huge work here!
Is it possible to merge it?
@luc-demeyer https://github.com/luc-demeyer  what about 8.0 PR that you have evoked in november?

thanks

—
Reply to this email directly or view it on GitHub https://github.com/OCA/account-financial-tools/pull/51#issuecomment-69190647 .  https://github.com/notifications/beacon/AHWZQeMJlB35qS4_3VHVW57KD_uod1Euks5nfpNsgaJpZM4CpKrQ.gif

lepistone commented:
Thanks for the work!

I am happy to merge this, but the diff seems huge. Is it against the correct branch?

sbidoul commented:
Normally it's "just" two new (large) modules.

@lepistone if you could merge #122, then @luc-demeyer can rebase and it should be green and you can merge this one.

lepistone commented:
Hi @sbidoul @luc-demeyer I merged #122 and restarted the build here.

rdeheele commented:
travis fails on test_3:
value_depreciated is not equal to 325.08

sbidoul commented:
Argh. That's strange this test was ok a few days ago.

lepistone commented:
maybe something else due to the new year?

rdeheele commented:
compute_depreciation_board method creates automatically a new depreciation line at the end of 2014 year.
Therefore, value depreciated is now 328.08 + 666.6 = 991.68

lepistone commented:
while fixing that, can you please add some more words in the function name and docstring of the test about what is going on? Something like "we enter a manual depreciation line for 2013, and one for 2014 should be automatically created". Thanks!

sbidoul committed:
[FIX] account_asset_management: fix test suite for 2015

It broke because the module generates init entries for dates
where there is no fiscal period.

sbidoul committed:
[IMP] account_asset_management: improve test suite comments

sbidoul commented:
@luc-demeyer @rdeheele @lepistone The PR to fix the test suite is there: https://github.com/luc-demeyer/account-financial-tools/pull/8

lepistone commented:
@sbidoul 's branch LGTM, thanks

luc-demeyer committed:
Merge pull request #8 from acsone/7.0-fix-asset-management-test-suite-2015-sbi

7.0 make asset management test suite FY independent

sbidoul commented:
@lepistone you can merge now

lepistone commented:
:trophy:

sbidoul commented:
Great job all!

@luc-demeyer for 8.0, can you do it the following way in order to preserve history and keep the merge point between 7.0 and 8.0 (not the exact commands)
- git checkout 8.0
- git merge 7.0
- add your commits to adapt for 8.0 if any
- create a new PR or force push on the branch of #52 

If you prefer, you can send me your diff with the 8.0 changes and I'll do the git incantation for you.

rdeheele commented:
hello @luc-demeyer , all my support to push the 8.0 branch ;)

luc-demeyer commented:
@rdeheele 
Can you first merge https://github.com/OCA/account-financial-tools/pull/129 ?
Once this is done, I'll merge the V7 asset mgt work into V8.

rdeheele commented:
hello @luc-demeyer , #129 is merged.

luc-demeyer commented:
@rdeheele 
Romain, I'll need some help. I pushed the V8 version but Travis fails for reasons outside my commit.

Cf. travis log:

  File "/home/travis/odoo-8.0/openerp/models.py", line 48, in <module>
    import pytz
ImportError: No module named pytz
